tests/robustness/validate/watch.go

```
@@ -76,17 +77,37 @@ func validateWatch(lg *zap.Logger, cfg Config, reports []report.ClientReport, ev
 	return nil
 }
 
-func validateBookmarkable(lg *zap.Logger, report report.ClientReport) (err error) {
+func validateBookmarkable(lg *zap.Logger, eventHistory []model.PersistedEvent, report report.ClientReport) (err error) {
 	for _, op := range report.Watch {
 		var lastProgressNotifyRevision int64
+		var gotEventBeforeProgressNotify *model.PersistedEvent
 		for _, resp := range op.Responses {
 			for _, event := range resp.Events {
 				if event.Revision <= lastProgressNotifyRevision {
 					lg.Error("Broke watch guarantee", zap.String("guarantee", "bookmarkable"), zap.Int("client", report.ClientID), zap.Int64("revision", event.Revision))
 					err = errBrokeBookmarkable
```

Should we return the error here?

---

No, we want to log all the errors.